File: lc_py/lcPy/sliding_window/CountSubarrays2302.py
class CountSubarrays2302:
    def countSubarrays(self, nums: list[int], k: int) -> int:
        n = len(nums)
        left = 0
        prefix_sum = 0
        ans = 0

        for right in range(n):
            prefix_sum += nums[right]
            while left <= right and prefix_sum * (right - left + 1) >= k:
                prefix_sum -= nums[left]
                left += 1
            ans += right - left + 1

        return ans

# 测试
sol = Solution()
nums = [2, 1, 4, 3, 5]
k = 10
print(sol.CountSubarrays2302(nums, k))

File: lc_java/lcJava/slidingWindows/CountSubarrays2302.java
package slidingWindows;


class CountSubarrays2302{
    public long countSubarrays(int[] nums, long k) {
        int n = nums.length;
        int left = 0;
        long prefixSum = 0, ans = 0;
        
        for (int right = 0; right < n; ++right) {
            prefixSum += nums[right];
            while (left <= right && prefixSum * (right - left + 1) >= k) {
                prefixSum -= nums[left++];
            }
            ans += right - left + 1;
        }
        return ans;
    }
    
    public static void main(String[] args) {
        CountSubarrays2302 sol = new CountSubarrays2302();
        int[] nums = {2,1,4,3,5};
        long k = 10;
        System.out.println(sol.countSubarrays(nums, k));
    }
}